packages/modules-sdk/src/remote-query.ts

```diff
@@ -86,34 +86,40 @@ export class RemoteQuery {
     relations: string[]
     args: Record<string, unknown[]>
   } {
+    const dataCopy = JSON.parse(JSON.stringify(data))
+
     let fields: Set<string> = new Set()
     let relations: string[] = []
+    let selectAll = false
 
-    data.fields?.forEach((field: string) => {
-      if (field === "*") {
-        // Select all, so we don't specify any field and rely on relation only
+    dataCopy.fields?.forEach((field: string) => {
+      if (selectAll || field === "*") {
+        selectAll = true
```

Comment:
Do we need `selectAll` or should we just do `dataCopy.fields = []` here?

Reply:
<img width="910" alt="Screenshot 2024-03-19 at 17 14 39" src="https://github.com/medusajs/medusa/assets/25098370/879ea603-d3bc-4536-98db-c0999d3370e6">

the only issue is that the loop still continue if you do that, but I can improve still

Reply:
<img width="1148" alt="Screenshot 2024-03-19 at 17 21 28" src="https://github.com/medusajs/medusa/assets/25098370/2e36007a-1854-434b-87d0-4e594d28ed1f">
Sorry I ve removed the comment without purpose. I was saying that you can't because you still go through the loop. But I can still improve though